src/Common/src/Microsoft/Win32/SafeHandles/SafeFileHandle.Unix.cs

```diff
@@ -35,7 +40,7 @@ internal static SafeFileHandle Open(string path, Interop.Sys.OpenFlags flags, in
             bool enoentDueToDirectory = (flags & Interop.Sys.OpenFlags.O_CREAT) != 0;
 
             // Open the file. 
-            int fd;
+            IntPtr fd;
```

Comment:
Should Open be defined to return a SafeFileHandle?  Then we could have an Interop.CheckIo that takes a SafeHandle instead of an IntPtr, and we could avoid doing the whole dance of instantiating the SafeFileHandle and then calling SetHandle on it subsequently.